src/VPolytope.jl

```diff
@@ -385,16 +386,25 @@ Return the polytope obtained by removing the redundant vertices of the given pol
                computations backend, see
                [Polyhedra's documentation](https://juliapolyhedra.github.io/Polyhedra.jl/latest/installation.html#Getting-Libraries-1)
                for further information
+- `solver`  -- (optional, default: `default_lp_solver(N)`) the linear programming
+               solver used in the backend
 
 ### Output
 
 A new polytope such that its vertices are the convex hull of the given polytope.
 """
 function remove_redundant_vertices(P::VPolytope{N};
-                                   backend=default_polyhedra_backend(P, N))::VPolytope{N} where {N<:Real}
+                                   backend=default_polyhedra_backend(P, N),
+                                   solver=default_lp_solver(N))::VPolytope{N} where {N<:Real}
     require(:Polyhedra; fun_name="remove_redundant_vertices")
+
     Q = polyhedron(P; backend=backend)
-    removevredundancy!(Q)
+    if Polyhedra.supportssolver(typeof(Q))
+        vQ = Polyhedra.vrep(Q)
+        Polyhedra.setvrep!(Q, Polyhedra.removevredundancy(vQ, solver))
```

Comment:
Does this accept `solver == nothing`? Would it not make more sense to catch this case and use the default LP solver?

Reply:
The `solver` option is mandatory in ` Polyhedra.removevredundancy(vQ, solver)`; the function that i intend to use here is [this one](https://github.com/JuliaPolyhedra/Polyhedra.jl/blob/master/src/redundancy.jl#L211). The argument in `remove_redundant_vertices` is a VPolytope and the logic is to either:

- attach the GLPK solver (by default, but any other LP solver supported by JuMP is ok)  if `Q` supports it and call the 2 argument version of `removevredundancy`

or

- do not attach any solver and use the solver's `removevredundancy!` function (line 406 below)

Reply:
This does not answer my question.

Reply:
> Does this accept solver == nothing?

What is "this"? The particular invocation `Polyhedra.removevredundancy`  in line 404 does not accept `nothing` as solver, that explains why i made the check in line 402.

> Would it not make more sense to catch this case and use the default LP solver?

Which case, if `solver` is nothing? It is not always problematic (for example if the backend is CDDLib, `solver` is not needed).

Reply:
> that explains why i made the check in line 402

Line 402 does not look at `solver`, so it cannot fix the problem.

> Which case, if solver is nothing?

Yes.

> It is not always problematic

This somewhat implies that it is problematic *sometimes*. We should solve these cases.

Reply:
The line number changed after some commit, i meant the check `if Polyhedra.supportssolver(typeof(Q))`.

Reply:
I know, but there is no `solver` in that line!

Reply:
> This somewhat implies that it is problematic sometimes. We should solve these cases

This function `remove_redundant_vertices` is an attempt to use `Polyhedra.removevredundancy` but with a `VPolytope` input, which is of course not understood by Polyhedra, so one has to make some decision regarding conversion to a type that is understood by it.

What we've done in `AbstractPolytope.jl` is to create a default polyhedron without solver, and pair to one solve later in this function (but this is just a choice, as solver can be stored in the type [here](https://github.com/JuliaPolyhedra/Polyhedra.jl/blob/master/src/defaultlibrary.jl#L9)). I think it is clear to do it this way because for certain applications in LazySets, the `solver` is not needed.